File: DroidsAPI/src/main/java/com/github/rochedo098/droidsapi/api/script/classes/listeners/RobotActionsListener.java
package com.github.rochedo098.droidsapi.api.script.classes.listeners;

import org.jetbrains.annotations.ApiStatus;

public interface RobotActionsListener {
    default void toUse(String blockPos) {}

    default void toBreak(String blockPos) {}

    @ApiStatus.Experimental
    default String checkBlock(String blockPos) { return "null"; }
}

File: src/main/kotlin/com/github/rochedo098/droids/api/script/classes/listeners/RobotActionsListener.kt
package com.github.rochedo098.droids.api.script.classes.listeners

import org.jetbrains.annotations.ApiStatus

interface RobotActionsListener {
    fun toUse(blockPos: String?) {}
    fun toBreak(blockPos: String?) {}

    @ApiStatus.Experimental
    fun checkBlock(blockPos: String?): String? {
        return "null"
    }
}